src/websocket/stream.py

```diff
@@ -210,53 +239,47 @@ async def transcribe_chunk_partial(
             text = ""
 
             if len(self.agreement.unconfirmed) > 0:
-                text = " ".join([w.word for w in new_words])
+                # hacky workaround for doubled word between finals
+                if len(new_words) > 0 and len(self.final_transcriptions) > 0:
+                    if new_words[0].word == self.final_transcriptions[-1]["result"][-1]["word"]:
+                        new_words.pop()
```

Comment:
shouldnt it pop from front instead of back?

- Also this checks only for one word, imagine the overhang is multiple?
- sometimes multiple words are grammatically correct like: `She realized that he had had enough`

Reply:
In theory yes, in practice unlikely